pkg/cmd/admin/policy/psp_review.go

```diff
@@ -0,0 +1,302 @@
+	}
+	switch {
+	case len(userOrSA) > 0 || len(o.Groups) > 0:
+		podSecurityPolicySubjectReview := &securityapi.PodSecurityPolicySubjectReview{
+			Spec: securityapi.PodSecurityPolicySubjectReviewSpec{
+				Template: *o.PodTemplateSpec,
+				User:     userOrSA,
+				Groups:   o.Groups,
+			},
+		}
+		response, err = o.subjectReviewClient.PodSecurityPolicySubjectReviews(o.Namespace).Create(podSecurityPolicySubjectReview)
+		if err != nil {
+			return err
+		}
+	default:
+		podSecurityPolicySelfSubjectReview := &securityapi.PodSecurityPolicySelfSubjectReview{
+			Spec: securityapi.PodSecurityPolicySelfSubjectReviewSpec{
+				Template: *o.PodTemplateSpec,
+			},
+		}
+		response, err = o.selfSubjectReviewClient.PodSecurityPolicySelfSubjectReviews(o.Namespace).Create(podSecurityPolicySelfSubjectReview)
+		if err != nil {
+			return err
+		}
+	}
+	return o.PrintObject(response, out)
+}
+
+func (o *subjectReviewOptions) Complete(f *clientcmd.Factory, args []string, cmd *cobra.Command) error {
+	if len(o.User) > 0 && len(o.ServiceAccount) > 0 {
+		return fmt.Errorf("--user and --serviceaccount are mutually exclusive")
+	}
+	namespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.Namespace = namespace
+	oclient, _, err := f.Clients()
+	if err != nil {
+		return fmt.Errorf("unable to obtain client: %v", err)
+	}
+	o.subjectReviewClient = oclient
+	o.selfSubjectReviewClient = oclient
+
+	o.Namespace = namespace
+	mapper, typer := f.Object()
+	infos, err := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+		NamespaceParam(namespace).
+		FilenameParam(enforceNamespace, &resource.FilenameOptions{Recursive: false, Filenames: []string{o.Filename}}).
+		SingleResourceType().
+		Do().
+		Infos()
+
+	if err != nil {
+		return fmt.Errorf("unable to find a resource: %v", err)
+	}
+	if len(infos) != 1 {
+		return fmt.Errorf("expected a resource for path: %s", o.Filename)
+	}
+	o.PodTemplateSpec, err = returnPodTemplateSpec(infos[0].Object)
+	if err != nil {
+		return fmt.Errorf("unable to find podTemplateSpec from %s: %v", o.Filename, err)
+	}
+
+	o.PrintObject = func(c *cobra.Command, m meta.RESTMapper) func(runtime.Object, io.Writer) error {
+		return func(obj runtime.Object, out io.Writer) error {
+			return f.PrintObject(c, m, obj, out)
+		}
+	}(cmd, mapper)
+
+	return nil
+}
+
+func NewCmdSubjectReview(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &subjectReviewOptions{}
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s", name),
+		Long:    subjectReviewLong,
+		Short:   "Check whether a user or a ServiceAccount can create a Pod.",
+		Example: fmt.Sprintf(subjectReviewExamples, fullName, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := o.Complete(f, args, cmd); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
+			}
+			err := o.Run(out)
+			kcmdutil.CheckErr(err)
```

Comment:
The body of this method should go towards:
```
kcmdutil.CheckErr(opts.Complete(f, cmd, args, out))
kcmdutil.CheckErr(opts.Validate())
kcmdutil.CheckErr(opts.Run())
```
Not sure if you need `Validate`, yet ;)